src/pages/StatusPage/index.tsx

```diff
@@ -3,69 +3,86 @@ import { PageStatus, StatusRow } from '../../components';
 import { begin } from '../../services';
 import { Api, ApiWithResponse, Response } from '../../typings/Api';
 import { Container } from './StatusPage.elements';
-
+ 
 interface StatusPageProps {
-  /** An array of API objects */
-  apis: Api[];
-  /** The interval at which to call the APIs in milliseconds */
-  interval?: number;
+ /** An array of API objects */
+ apis: Api[];
+ /** The interval at which to call the APIs in milliseconds */
+ interval?: number;
 }
-
+ 
 interface ApiWithError extends ApiWithResponse {
-  hasError: boolean;
+ hasError: boolean;
 }
-
+ 
 interface StatusPageState {
-  apis: ApiWithError[];
-  hasError: boolean;
+ apis: ApiWithError[];
+ hasError: boolean;
 }
-
+ 
 class StatusPage extends Component<StatusPageProps> {
-  public state: StatusPageState = {
-    apis: [],
-    hasError: false,
-  };
-
-  public handleError = (api: Api, response: Response) => {
-    const apis = [...this.state.apis];
-    const index = apis.findIndex(stateApi => stateApi.api === api);
+ public state: StatusPageState = {
+   apis: [],
+   hasError: false,
+ };
+ 
+ 
+ public handleResponse = (api: Api, response: Response) => {
+   console.log('handleResponse' + response);
+   const apis = [...this.state.apis];
+   const index = apis.findIndex(stateApi => stateApi.api === api);
+   apis[index].response = response;
+  
+   if (response.code == 200) {
```

Comment:
There are a variety of responses that can indicate OK, more than just the standard 200. Would it make sense to check a range of codes or to just make the consumer adhere to returning a 200 to indicate everything's good?

Reply:
I change according to this link : https://www.apiscience.com/blog/javascript-validate-api-response-codes/

Reply:
Perfect!